pypeit/reduce.py

```diff
@@ -394,6 +386,80 @@ def run(self, ra=None, dec=None, obstime=None, std_trace=None, show_peaks=False,
             else:
                 self.global_sky = self.global_skysub(skymask=self.skymask, show=self.reduce_show)
 
+        return self.global_sky, self.sobjs_obj, self.skymask
+
+    def run_extraction(self, global_sky, sobjs_obj, skymask, ra=None, dec=None, obstime=None, return_negative=False):
+        """
+        Primary code flow for PypeIt reductions
+
+        *NOT* used by COADD2D
+
+        Args:
+            global_sky (`np.ndarray`_):
+                Initial global sky model
+            sobjs_obj (:class:`pypeit.specobjs.SpecObjs`):
+                List of objects found during `run_objfind`
+            skymask (`np.ndarray`_):
+               Boolean image indicating which pixels are useful for global sky subtraction
+            ra (str, optional):
```

Comment:
Can we avoid `str` for ra, dec?  
These should only be float and decimal deg in PypeIt.